### PR TITLE
Add check for startPos to fix #7864

### LINKED ISSUE #7864: Marker element click event throws: Uncaught TypeError: Cannot read property 'x' of null
**mapbox-gl-js version**: 0.52.0

**browser**: macOS Chrome Version 71.0.3578.98 (Official Build) (64-bit)

### Steps to Trigger Behavior

 1. add a html marker;
 2. add event by addEventListener to marker's html dom;
 3. fire the event, throws the error: Uncaught TypeError: Cannot read property 'x' of null

### Link to Demonstration

https://jsfiddle.net/ParryQiu/hnkwqfue/12/

console out put error:

```
vendor.js:51848 ERROR TypeError: Cannot read property 'x' of null
    at a.equals (vendor.js:18268)
    at HTMLDivElement.<anonymous> (vendor.js:18272)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (polyfills.js:2745)
    at Object.onInvokeTask (vendor.js:53986)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (polyfills.js:2744)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.runTask (polyfills.js:2517)
    at ZoneTask.push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask [as invoke] (polyfills.js:2820)
    at invokeTask (polyfills.js:4066)
    at HTMLDivElement.globalZoneAwareCallback (polyfills.js:4092)
    at Map.<anonymous> (main.js:7314)
```

### PR DESCRIPTION
**Pull request by msbarry**
Add check for `startPos` to fix #7864.

This falls back to ignoring click tolerance and just firing the click event in the case where startPos is not defined. I've been running with it in production for a few months and have not seen the error since then.

### CODE REVIEW

File: test/unit/ui/map_events.test.js [hunk 1]
```diff
@@ -585,3 +585,19 @@ test(`Map#on mousedown does not fire subsequent click event if mouse position ch
...
+
+test(`Map#on click fires subsequent click event if there is no corresponding mousedown/mouseup event`, (t) => {
+    const map = createMap(t, { clickTolerance: 4 });
+
+    const click = t.spy();
+    map.on('click', click);
+    const canvas = map.getCanvas();
+
+    const MouseEvent = window(canvas).MouseEvent;
+    const event = new MouseEvent('click', {bubbles: true, clientX: 100, clientY: 100});
```
Comment (msbarry) [inline test/unit/ui/map_events.test.js]: Alternatively we could expose a `simulate.fakeClick` utility, but I didn't think we'd want to encourage anyone to re-use that since it only applies to a corner case.
